packages/dhis2/test/metadata/lookup.test.js

```diff
@@ -0,0 +1,25 @@
+import { expect } from 'chai';
+import jp from 'jsonpath';
+import path from 'node:path';
+import data from '../../src/meta/data/metadata.json' assert { type: 'json' };
+import extractLookups from '@openfn/parse-jsdoc';
+
+let queries;
+
+before(async () => {
+  // Parse Adaptor.js and pull out all of its lookup queries
+  queries = await extractLookups(path.resolve('src/Adaptor.js'));
+});
+
+describe('Salesforce lookup tests', async () => {
+  // Unit tests of each query against a sample model
+  describe('create', () => {
+    it('sObject: should list resourceTypes', () => {
+      const results = jp.query(data, queries.create.resourceType);
+      expect(results).to.have.lengthOf(4);
+      expect(results[0]).to.equal('trackedEntityInstances');
+    });
+
+    // TODO what about testing the lookups on the typescript though?
```

Comment:
Uh turns out this is a can of worms.

Having to parse a .ts file and pull out the docs on the property of on object is some heavy stuff.

jsddoc2md won't parse ts files (not unreasonably!)

There is a tsdoc project but it is concerned only with functions, not objects/interfaces.

TS does parse the jsdoc on the property definition - I know this because I use it in monaco. Bit worried we need to use the ts compiler directly to pull out the right information...

Reply:
I may defer this as and spin out an issue. It's a big deal and probably shouldn't be a blocker to magic functions.